src/csharp/Grpc.Core.Api/CallOptions.cs

```diff
@@ -47,14 +47,34 @@ public struct CallOptions
         /// <param name="credentials">Credentials to use for this call.</param>
         public CallOptions(Metadata headers = null, DateTime? deadline = null, CancellationToken cancellationToken = default(CancellationToken),
                            WriteOptions writeOptions = null, ContextPropagationToken propagationToken = null, CallCredentials credentials = null)
+            : this(waitForReady: false, headers, deadline, cancellationToken, writeOptions, propagationToken, credentials)
+        {
+        }
+
+        /// <summary>
+        /// Creates a new instance of <c>CallOptions</c> struct.
+        /// </summary>
+        /// <param name="headers">Headers to be sent with the call.</param>
+        /// <param name="deadline">Deadline for the call to finish. null means no deadline.</param>
+        /// <param name="cancellationToken">Can be used to request cancellation of the call.</param>
+        /// <param name="writeOptions">Write options that will be used for this call.</param>
+        /// <param name="propagationToken">Context propagation token obtained from <see cref="ServerCallContext"/>.</param>
+        /// <param name="credentials">Credentials to use for this call.</param>
+        /// <param name="waitForReady">
+        /// If <c>true</c> and channel is in <c>ChannelState.TransientFailure</c>, the call will attempt waiting for the channel to recover
+        /// instead of failing immediately (which is the default "FailFast" semantics).
+        /// Note: experimental API that can change or be removed without any prior notice.
+        /// </param>
+        public CallOptions(bool waitForReady, Metadata headers = null, DateTime? deadline = null, CancellationToken cancellationToken = default(CancellationToken),
```

Comment:
it seems odd that the `waitForReady`  argument is the only one that's not optional. Also, the ordering of the arguments seems strange (why is waitForReady first?). I understand you somehow need to disambiguate the two constructors, but I'm not sure this is the best way of doing so.

Reply:
As mentioned in the top comment, it needs to be non-optional to disambiguate which method is being called.

And it needs to be first because non-optional parameters must come before optional parameters.